common/src/main/java/org/figuramc/figura/model/rendering/ImmediateAvatarRenderer.java

```diff
@@ -542,32 +529,81 @@ private void pushToBuffer(int faceCount, VertexData vertexData, PartCustomizatio
 
         int overlay = customization.overlay;
         int light = vertexData.fullBright ? LightTexture.FULL_BRIGHT : customization.light;
+        ToBeConsumedVertexData[] vertexDatas = new ToBeConsumedVertexData[vertCount]; // why the fuck did i use duckai for this
+        for (int i = 0; i < vertCount; i++) {
+            Vertex vertex = vertices.get(i);
+
+            pos.set(vertex.x, vertex.y, vertex.z, 1);
+            pos.transform(customization.positionMatrix);
+            pos.add(pos.normalized().scale(vertexData.vertexOffset));
+            normal.set(vertex.nx, vertex.ny, vertex.nz);
+            normal.transform(customization.normalMatrix);
+            uv.set(vertex.u, vertex.v, 1);
+            uv.divide(uvFixer);
+            uv.transform(customization.uvMatrix);
+
+            vertexDatas[i] = ToBeConsumedVertexData.get(this,
+                pos,
+                vertexData.color, customization.alpha, // Yes I know this is supposed to be a static variable but shhhhh
+                uv,
+                overlay,
+                light,
+                normal
+            );
 
+        }
         VERTEX_BUFFER.getBufferFor(vertexData.renderType, vertexData.primary, vertexConsumer -> {
             for (int i = 0; i < vertCount; i++) {
-                Vertex vertex = vertices.get(i);
-
-                pos.set(vertex.x, vertex.y, vertex.z, 1);
-                pos.transform(customization.positionMatrix);
-                pos.add(pos.normalized().scale(vertexData.vertexOffset));
-                normal.set(vertex.nx, vertex.ny, vertex.nz);
-                normal.transform(customization.normalMatrix);
-                uv.set(vertex.u, vertex.v, 1);
-                uv.divide(uvFixer);
-                uv.transform(customization.uvMatrix);
-
-                vertexConsumer
-                        .vertex(pos.x, pos.y, pos.z)
-                        .color((float) vertexData.color.x, (float) vertexData.color.y, (float) vertexData.color.z, customization.alpha)
-                        .uv((float) uv.x, (float) uv.y)
-                        .overlayCoords(overlay)
-                        .uv2(light)
-                        .normal((float) normal.x, (float) normal.y, (float) normal.z)
-                        .endVertex();
+                ToBeConsumedVertexData data = vertexDatas[i];
+                data.consume(this,vertexConsumer);
             }
         });
     }
 
+    private static class ToBeConsumedVertexData {
+        public Float x,y,z;
+        public Float r,g,b,a;
+        public Float uvX,uvY;
+        public int overlay;
+        public int light;
+        public Float nX,nY,nZ;
+        public boolean consumed;
```

Comment:
any reason these are capital `Float` (the box type) instead of lowercase `float` (the primitive)? none of them look nullable to me

Reply:
Typo, Used to other languages using Float instead of float